lib/websock.ex

```diff
@@ -24,19 +24,26 @@ defmodule WebSock do
   @typedoc "All possible frame types"
   @type opcode :: data_opcode() | control_opcode()
 
+  @typedoc "The structure of an outbound message"
+  @type message :: {opcode(), payload()}
+
+  @typedoc "Convenience type for one or many messages"
+  @type messages :: message() | [message()]
+
   @typedoc "The result as returned from init, handle_in, handle_control & handle_info calls"
   @type handle_result ::
-          {:push, {opcode(), message()} | [{opcode(), message()}], state()}
-          | {:reply, term(), {opcode(), message()} | [{opcode(), message()}], state()}
+          {:push, messages(), state()}
+          | {:reply, term(), messages(), state()}
           | {:ok, state()}
           | {:stop, {:shutdown, :restart} | term(), state()}
           | {:stop, term(), close_detail(), state()}
+          | {:stop, term(), close_detail(), messages(), state()}
```

Comment:
This is the salient change in this PR; the rest of the noise here is just to clean up typespecs